controlpanel/api/kubernetes.py

```diff
@@ -9,10 +9,30 @@
 class KubernetesClient:
     """
     Wraps kubernetes.client default configuration with currently logged-in
-    user's credentials
+    user's credentials unless `use_cpanel_creds=True`, in that case the
+    in-cluster (or ~/.kube/config) credentials are used.
+
+    **IMPORTANT**: Do not pass `use_cpanel_creds=True` unless strictly
+    necessary. The CP `ServiceAccount` has more privileges than normal users.
+    For most operations performed by a user use their ID token to make requests
+    to the k8s API.
     """
 
-    def __init__(self, id_token=None):
+    def __init__(self, id_token=None, use_cpanel_creds=False):
+        if not use_cpanel_creds and not id_token:
+            raise ValueError(
+                "please provide an id_token (preferred) or pass use_cpanel_creds"
+                "=True (this would cause the use of the host credentials so "
+                "be careful when using it)"
+            )
+
+        if id_token and use_cpanel_creds:
+            raise ValueError(
+                "id_token and use_cpanel_creds can't be used together. "
+                "Remember: avoid using the Control Panel credentials to talk with "
+                "the k8s API unless stricly necessary."
+            )
+
```

Comment:
I think you've forgotten to push some code here

Reply:
Oh god, you're absolutely right! Superstar for catching this @andyhd 🌟❤️ 

I was tweaking this class and eventually decided to move the refactoring experiment into a separate PR but I thrown away the feature with the dirty water 🤦‍♂ 

The missing code is simply a `and not use_cpanel_creds` added to the `if`:

```python
        if settings.ENABLED['k8s_rbac'] and not use_cpanel_creds:
            if id_token is None:
                # ...
```

as in that case we don't want to use the `id_token`.

I'll dig this out or add again. Amazing, thanks again for spotting this Andy ❤️